cql3/query_options.cc

```diff
@@ -135,12 +135,21 @@ void query_options::prepare(const std::vector<lw_shared_ptr<column_specification
     ordered_values.reserve(specs.size());
     for (auto&& spec : specs) {
         auto& spec_name = spec->name->text();
+        bool found_value_for_name = false;
         for (size_t j = 0; j < names.size(); j++) {
             if (names[j] == spec_name) {
```

Comment:
Quadratic complexity detected!

Reply:
The binary protocol has an [interesting comment ](https://github.com/apache/cassandra/blob/a07c15d72d575845624746b828067ed8d2dc3d9d/doc/native_protocol_v4.spec#L382):

> for EXECUTE statements,
              since the names for the expected values was returned during preparation,
              a client can always provide values in the right order without any names
              and using this flag, while supported, is almost surely inefficient.

looks like this could be optimized at the driver side, do drivers actually do this? Rust one doesn't (yet).

I guess it would be easy to change it to a hashmap, but would this even be faster for a few values?